Tests/SwerverTests/RouterSpec/RoutesSpec.swift

```diff
@@ -9,28 +9,57 @@ class RoutesSpec: QuickSpec {
         describe("A Routes") {
             var routes = Routes()
             let targetRoute = URL(string: Resource.test.rawValue)
+            let invalidURL = URL(string: String())
+            let unknownURL = URL(string: Resource.not_there.rawValue)
             
             it("accept and store new routes") {
-                routes.addRoute(url: targetRoute!, actions: [RequestMethod.get: GetAction(), RequestMethod.head: HeadAction()])
-                expect(routes.routeExist(url: targetRoute!)).to(beTrue())
+                routes.addRoute(url: targetRoute, actions: [RequestMethod.get: GetAction(), RequestMethod.head: HeadAction()])
+                expect(routes.routeExist(url: targetRoute)).to(beTrue())
             }
             
-            it("can return an options action object with all the allowed methods for an existing route") {
-                let optionsAction = OptionsAction()
-                optionsAction.setAllowedMethods(methods: ["GET", "HEAD"])
-                
-                let expectedOptionsAction = routes.optionsAction(url: targetRoute!)
-
-                expect(expectedOptionsAction.dispatch()).to(equal(optionsAction.dispatch()))
+            it("can return the total number of routes") {
+                expect(routes.numberOfRoute()).to(equal(1))
             }
             
-            it("can return a dictionary of all allowed methods for an existing route") {
-                let expectedListOfMethods = routes.fetchAllActions(url: targetRoute!)
-                let expectedActionType =
+            it("can return a dictionary of all allowed methods for an existing route with the request method as keys and http action as values") {
+                let expectedDictOfMethods = routes.fetchAllActions(url: targetRoute)
                 
+                expect(expectedDictOfMethods.count).to(equal(2))
+                expect(expectedDictOfMethods.keys.contains(RequestMethod.get)).to(beTrue())
+                expect(expectedDictOfMethods.keys.contains(RequestMethod.head)).to(beTrue())
+                expect(expectedDictOfMethods[RequestMethod.get]).toNot(beNil())
```

Comment:
Can you make a better assertion than "it is not nil"?

Reply:
will look into that

Reply:
fixed. it'll expect the type of Optional<HttpAction> as value